### PR TITLE
Implement hashing and equality for Records

### PR DESCRIPTION
**Pull request by harryscholes**
## Types of changes

This PR implements the following changes:
_(Please tick any or all of the following that are applicable)_

* [x] :sparkles: New feature (A non-breaking change which adds functionality).
* [ ] :bug: Bug fix (A non-breaking change, which fixes an issue).
* [ ] :boom: Breaking change (fix or feature that would cause existing functionality to change).

## :clipboard: Additional detail

- If you have implemented new features or behaviour
  - **Provide a description of the addition** in as many details as possible.
Implement `hash` and `==` for `FASTA.Record` and `FASTQ.Record`.
  - **Provide justification of the addition**.

  - **Provide a runnable example of use of your addition**. This lets reviewers
    and others try out the feature before it is merged or makes it's way to release.
```julia
using FASTX

record = FASTA.Record("id", "desc", "ACGT")
record2 = FASTA.Record("id", "desc", "ACGT")
record1 == record2
hash(record1) == hash(record2)
unique([record1, record1, record2, record2]) == [record1] == [record2]

record = FASTQ.Record("id", "desc", "AAGCT", collect("@BCFF"))
record2 = FASTQ.Record("id", "desc", "AAGCT", collect("@BCFF"))
record1 == record2
hash(record1) == hash(record2)
unique([record1, record1, record2, record2]) == [record1] == [record2]
```
- If you have changed current behaviour...
  - **Describe the behaviour prior to you changes**
Identical `Record`s did not hash to the same value because they are implemented as `mutable struct`s.
  - **Describe the behaviour after your changes** and justify why you have made the changes,
    Please describe any breakages you anticipate as a result of these changes.
Identical `Record`s hash to the same value.

## :ballot_box_with_check: Checklist

- [x] :art: The changes implemented is consistent with the [julia style guide](https://docs.julialang.org/en/stable/manual/style-guide/).
- [ ] :blue_book: I have updated and added relevant docstrings, in a manner consistent with the [documentation styleguide](https://docs.julialang.org/en/stable/manual/documentation/).
- [ ] :blue_book: I have added or updated relevant user and developer manuals/documentation in `docs/src/`.
- [x] :ok: There are unit tests that cover the code changes I have made.
- [x] :ok: The unit tests cover my code changes AND they pass.
- [ ] :pencil: I have added an entry to the `[UNRELEASED]` section of the manually curated `CHANGELOG.md` file for this repository.
- [x] :ok: All changes should be compatible with the latest stable version of Julia.
- [ ] :thought_balloon: I have commented liberally for any complex pieces of internal code.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -284,6 +289,12 @@ end
         @test FASTQ.hasquality(record)
         @test FASTQ.quality(record) == b"@BCFFFDFHHHHHJJJIJIJJIJJJJJJJJIJJJJIIIJJJIJJJ" .- 33
 
+        record = FASTQ.Record("id", "desc", "AAGCT", collect("@BCFF"))
```
Comment (kescobo) [inline test/runtests.jl]: This should be `record1` I think...

File: test/runtests.jl [hunk 2]
```diff
@@ -36,7 +36,12 @@ import BioSequences:
         @test FASTA.sequence(record, 2:3) == dna"CG"
         @test FASTA.sequence(String, record) == "ACGT"
         @test FASTA.sequence(String, record, 2:3) == "CG"
-        @test record == FASTA.Record(">foo\nACGT\n")
+
+        record1 = FASTA.Record("id", "desc", "AGCT")
+        record2 = FASTA.Record("id", "desc", "AGCT")
```
Comment (kescobo) [inline test/runtests.jl]: Can you also add some records that lack some fields (esp description), or have differences, and add tests that check for inequality?

File: test/runtests.jl [hunk 3]
```diff
@@ -284,6 +289,12 @@ end
         @test FASTQ.hasquality(record)
         @test FASTQ.quality(record) == b"@BCFFFDFHHHHHJJJIJIJJIJJJJJJJJIJJJJIIIJJJIJJJ" .- 33
 
+        record = FASTQ.Record("id", "desc", "AAGCT", collect("@BCFF"))
+        record2 = FASTQ.Record("id", "desc", "AAGCT", collect("@BCFF"))
```
Comment (kescobo) [inline test/runtests.jl]: Same comment as above, add some tests for inequality, and include records that lack a description.

**Comment by kescobo** [comment]
Looks pretty good to me. There's one typo in the tests, and I've requested a couple of additional tests for inequality, but this is a great addition. Thanks!

File: test/runtests.jl (hunk 1 above)
Comment (harryscholes) [reply test/runtests.jl]: Whoops. Last minute change on my part to rename the variable

**Comment by harryscholes** [comment]
Thanks for your review and great ideas, Kevin. I've added those additional tests now

**Comment by kescobo** [comment]
Is there a way to hit [this line](https://codecov.io/gh/BioJulia/FASTX.jl/compare/d718b47f1f1938972d2012ccacedc90a525def4b...b2d056f90b908c32db98764e3ce06fde86d7c34e/diff#D2-92) with a test?

**Comment by kescobo** [comment]
👍 Awesome! I don't seem to have merge privileges on this repo, but @BenJWard I've reviewed this and think it looks good.

**Comment by BenJWard** [comment]
Thank you for prepping this @harryscholes, and thank's for taking the time to go over it @kescobo!

**Pull request merged**